File: rust/single_file/gcd.rs
fn main() {
    let var_a = 60;
    let var_b = 3420;
    println!("This is the result: {}", gcd(var_a, var_b));
}

fn gcd(a: u32, b: u32) -> u32 {
    if a % b == 0 { b } else { gcd(b, a % b) }
}

File: clang/gcd.c
#include <stdio.h>

int gcd(const int* const a,const int* const b) {
    if (*a % *b == 0) {
        return *b;
    } else {
        const int c = *a % *b;
        return gcd(b, &c);
    }
}

int main() {
    int num_one = 30;
    int num_two = 60;
    printf("This is the result: %d\n", gcd(&num_one, &num_two));
}
